app/controllers/users_controller.rb

```diff
@@ -1,11 +1,21 @@
 class UsersController < ApplicationController
+
   load_and_authorize_resource
   def index
     # @users = User.all
-    @users = User.order(:name).page params[:page]
+    unless params[:type].nil?
+      # This very... different implementation of displaying the users when a filter parameter is added is due to the
+      # way Rails converts a set of records to a traditional Ruby array when using the select method. Kaminari
+      # requires this syntax when paginating a normal array.
+      filtered_users = User.select{ |user| user.has_role? params[:type]}.sort_by(&:name)
+      @users = Kaminari.paginate_array(filtered_users).page params[:page]
+    else
+      @users = User.order(:name).page params[:page]
+    end
   end
 
   def update
+    # TODO: Rewrite this update method. It is poorly designed and possible vulnerable to attack.
```

Comment:
Yikes. Have we created an issue for addressing this yet? If not, be sure to sketch out the rough shape of the vulnerability. I kind of see where the issue might be with checking for `Admin` and `Instructor` via `include?` but to some degree, this should be protected via CSRF tokens.

Reply:
Looking at it now, you're right; I think CSRF tokens and our Rollify setup would prevent any severe security issues. I can't remember exactly why I left this comment but I think it had something to do with it being a simple ``include?`` check on a parameter that a user has full control to pass whatever they want. We can probably file this one under "things we can leave for now but should probably fix when we rebuild the app's authorization".